src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.cs

```diff
@@ -103,7 +103,7 @@ internal sealed partial class LocalRewriter : BoundTreeRewriterWithStackGuard
                     dynamicAnalysisSpans = dynamicInstrumenter.DynamicAnalysisSpans;
                 }
 
-                AssertValidLocalRewriting(loweredStatement);
+                LocalRewritingValidator.Validate(loweredStatement);
```

Comment:
`#if DEBUG`

Reply:
That method is conditional, which I think has the same effect.